Source/WTF/wtf/MallocSpan.h

```diff
@@ -67,33 +67,33 @@ template<typename T, typename Malloc = FastMalloc> class MallocSpan {
 
     static MallocSpan malloc(size_t size)
     {
-        return MallocSpan { static_cast<T*>(Malloc::malloc(size)), size };
+        return MallocSpan { static_cast<T*>(Malloc::malloc(size * sizeof(T))), size };
```

Comment:
Probably it would be a good idea to use `wtf/CheckedArithmetic.h` to ensure that the multiplication doesn't overflow, something like:

```suggestion
        Checked<size_t> allocationSize = sizeof(T);
        allocationSize *= size; 
        return MallocSpan { static_cast<T*>(Malloc::malloc(allocationSize), size };
```